community/kernel/src/main/java/org/neo4j/kernel/impl/locking/SimpleStatementLocksFactory.java

```diff
@@ -19,11 +19,36 @@
  */
 package org.neo4j.kernel.impl.locking;
 
+import org.neo4j.kernel.configuration.Config;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A {@link StatementLocksFactory} that creates {@link SimpleStatementLocks}.
+ */
 public class SimpleStatementLocksFactory implements StatementLocksFactory
 {
+    private final Locks locks;
+
+    public SimpleStatementLocksFactory( Locks locks )
```

Comment:
Initialisation lifecycle of `SimpleStatementLocksFactory` is a bit different then any other `StatementLocksFactory` which is a bit confusing. Can we please add default constructor and assign locks in `initialize` instead to much expected service behaviour, and mark `public SimpleStatementLocksFactory( Locks locks )` as test only for convenient test usage